tcms/core/tests/test_markdown.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+import unittest
+import bleach
+
+from tcms.core.templatetags.extra_filters import markdown2html
+
+
+class TestMarkdownExtraFilters(unittest.TestCase):
+
+    def setUp(self):
+        self.md_simple_text = markdown2html("__*hello!*__")
+        self.md_table = markdown2html("""|Stx |Desc |\n|----|-----|\n|Head|Title|\n|Text|Text |""")
+        self.md_nl2br = markdown2html("""Line 1
+Line 2""")
+        self.md_fenced_code = markdown2html("""```{
+"firstName": "John",
+"lastName": "Smith",
+"age": 25}``` """)
+
+    def test_markdown2html_convert_paragraphs(self):
+        self.assertEqual(self.md_simple_text, "<p><strong><em>hello!</em></strong></p>")
+
+    def test_markdown2html_convert_tables(self):
+        self.assertEqual(self.md_table, """<table>

```

Comment:
i think you can do:
```python
        self.md_table = """|Stx |Desc |\n|----|-----|\n|Head|Title|\n|Text|Text |"""
```
in the setup method and here:
``` python
    def test_markdown2html_convert_tables(self):
        self.assertEqual(markdown2html(self.md_table), """<table>
```
that would be clearer IMO

Reply:
Or you can pass the input arguments directly as string constants because we don't need to carry them around as class attributes.